src/doc_builder/build_embeddings.py

```diff
@@ -0,0 +1,439 @@
+                page_info["page"] = file.with_suffix("").relative_to(doc_folder).as_posix()
+                with open(file, "r", encoding="utf-8-sig") as reader:
+                    content = reader.read()
+                content = clean_md(content)
+                content = process_md(content, page_info)
+                autodoc_content = "\n\n".join(_re_autodoc_all.findall(content))
+                content = _re_autodoc_all.sub("", content)
+                autodoc_chunks, new_anchors, errors = create_autodoc_chunks(
+                    autodoc_content,
+                    package,
+                    return_anchors=True,
+                    page_info=page_info,
+                    version_tag_suffix=version_tag_suffix,
+                )
+                markdown_chunks = create_markdown_chunks(
+                    content,
+                    page_info=page_info,
+                )
+
+                # Make sure we clean up for next page.
+                del page_info["page"]
+
+                page_chunks = markdown_chunks + autodoc_chunks
+
+                if is_python_module:
+                    page_chunks = [
+                        chunk._replace(text=resolve_links_in_text(chunk.text, package, anchor_mapping, page_info))
+                        for chunk in page_chunks
+                    ]
+
+                chunks.extend(page_chunks)
+
+        except Exception as e:
+            raise type(e)(f"There was an error when converting {file} to chunks to embed.\n" + e.args[0]) from e
+
+        if new_anchors is not None:
+            page_name = str(file.with_suffix("").relative_to(doc_folder))
+            for anchor in new_anchors:
+                if isinstance(anchor, tuple):
+                    anchor_mapping.update(
+                        {a: f"{page_name}#{anchor[0]}" for a in anchor[1:] if a not in anchor_mapping}
+                    )
+                    anchor = anchor[0]
+                anchor_mapping[anchor] = page_name
+
+        if errors is not None:
+            all_errors.extend(errors)
+
+    if len(all_errors) > 0:
+        raise ValueError(
+            "The deployment of the documentation will fail because of the following errors:\n" + "\n".join(all_errors)
+        )
+
+    return chunks
+
+
+def call_embed_api(chunks: List[Chunk]) -> List[Embedding]:
+    """
+    Using https://huggingface.co/inference-endpoints with a text embedding model
+    """
+    batch_size = 32
+    chunks_len = len(chunks)
+    embeddings = []
+
+    def query(payload):
+        response = requests.post(API_URL, headers=headers, json=payload)
+        return response.json()
+
+    # warm up API call
+    output_warmup = query({"inputs": "Hello World!"})
+    if isinstance(output_warmup, dict) and "error" in output_warmup:
+        if output_warmup["error"] == "503 Service Unavailable":
+            print("Waking up Embedding Inference Endpoints. Retrying in 5 minutes")
+            time.sleep(300)  # 5 minutes
```

Comment:
would be more satisfying to receive an event when the endpoint is available, but i guess that's painful to do in Python

Reply:
does HF ednpoints provided events? cc: @philschmid

Reply:
There is a [`huggingface_hub.InferenceEndpoint`](https://huggingface.co/docs/huggingface_hub/guides/inference_endpoints) client in Python if you want but it does the same as you (wait for model to be available).

Reply:
If instead of an API url you have the name of an inference endpoint, you can do:

```py
from huggingface_hub import get_inference_endpoint

endpoint = get_inference_endpoint(name="...")
client = endpoint.wait().client
client.text_embedding(...)
```

Reply:
@mishig25 Little update on inference endpoint. Before the `.wait()` call, you must do `.resume()`. If the endpoint scaled down to zero or have been paused, it will be restarted. So the snippet should look like this:

```py
from huggingface_hub import get_inference_endpoint

endpoint = get_inference_endpoint(name="...")
client = endpoint.resume().wait().client
client.feature_extraction(...)
```